cache/fast_lru_cache.cc

```diff
@@ -9,26 +9,34 @@
 
 #include "cache/fast_lru_cache.h"
 
+#include <math.h>
+
 #include <cassert>
 #include <cstdint>
 #include <cstdio>
+#include <functional>
 
 #include "monitoring/perf_context_imp.h"
 #include "monitoring/statistics.h"
 #include "port/lang.h"
+#include "util/hash.h"
 #include "util/mutexlock.h"
 
-#define KEY_LENGTH \
-  16  // TODO(guido) Make use of this symbol in other parts of the source code
-      // (e.g., cache_key.h, cache_test.cc, etc.)
-
 namespace ROCKSDB_NAMESPACE {
 
 namespace fast_lru_cache {
 
+namespace {
+// Returns x % 2^{bits}.
+inline uint32_t BinaryMod(uint32_t x, uint32_t bits) {
```

Comment:
If bits > 32 , the following shift count will be negative. 
Add an assert(bits <= 32) ?